[DOCS] DOC-340 partition local installation guide

Rachel-Reverie opened the pull request:
Please annotate your PR title to describe what the PR does, then give a brief bulleted description of your PR below. PR titles should begin with [BUGFIX], [FEATURE],  [DOCS], or [MAINTENANCE]. If a new feature introduces breaking changes for the Great Expectations API or configuration files, please also add [BREAKING]. You can read about the tags in our [contributor checklist](https://docs.greatexpectations.io/docs/contributing/contributing_checklist).

Changes proposed in this pull request:
DOC-340
- Partitions each section of the local install guide into individual files in the `components_local` folder.
- Updates local.md to use the partitioned files for its content
- Updated the content of the gude
   - Moved prerequisites above the "Steps" header
   - Moved creating a venv into its own step


After submitting your PR, CI checks will run and @cla-bot will check for your CLA signature.

For a PR with nontrivial changes, we review with both design-centric and code-centric lenses.

In a design review, we aim to ensure that the PR is consistent with our relationship to the open source community, with our software architecture and abstractions, and with our users' needs and expectations. That review often starts well before a PR, for example in github issues or slack, so please link to relevant conversations in notes below to help reviewers understand and approve your PR more quickly (e.g. `closes #123`).

Previous Design Review notes:
-
-
-


### Definition of Done
Please delete options that are not relevant.

- [X] I have made corresponding changes to the documentation



Thank you for submitting!

Rachel-Reverie committed:
DOC-340
- Partitions each section of the local install guide into individual files in the `components_local` folder.
- Updates local.md to use the partitioned files for its content
- Updated the content of the gude
   - Moved prerequisites above the "Steps" header
   - Moved creating a venv into its own step

Rachel-Reverie committed:
DOC-340
- Correct typo in markdown link formatting

ghost commented:
<details>
<summary><b>👇 <a href="https://app.codesee.io/r/reviews?pr=5425&src=https%3A%2F%2Fgithub.com%2Fgreat-expectations%2Fgreat_expectations">Click on the image</a> for a new way to code review</b>
</summary>

  - Make big changes easier — review code in small groups of related files
  - Know where to start — see the whole change at a glance
  - Take a code tour — explore the change with an interactive tour
  - Make comments and review — all fully sync’ed with github

    [Try it now!](https://app.codesee.io/r/reviews?pr=5425&src=https%3A%2F%2Fgithub.com%2Fgreat-expectations%2Fgreat_expectations)
</details>

[![Review these changes using an interactive CodeSee Map](https://s3.us-east-2.amazonaws.com/maps.codesee.io/images/github/great-expectations/great_expectations/5425/36141be8/a9c61b8f55728921914d032e3e577a1458cdb659.svg)](https://app.codesee.io/r/reviews?pr=5425&src=https%3A%2F%2Fgithub.com%2Fgreat-expectations%2Fgreat_expectations)

#### Legend
![CodeSee Map Legend](https://app.codesee.io/Legend_v2.png)

Rachel-Reverie committed:
DOC-340
- edit to make console input titles consistent

rdodev commented on docs/guides/setup/installation/components_local/_check_python_version.mdx:
````diff
@@ -0,0 +1,21 @@
+
+```console title="Terminal command"
+python3 --version
+```
````
Maybe worth adding an admonition re "as of this writing, GE requires a minimum version 3.x.x for python" but you know written good. Though that would add the annoyance of having to change yet another file every time GE bumps version support.

Rachel-Reverie replied:
We could get around that annoyance by having the statement be in an imported file, and then re-using that file in all of the places where we would make this statement.  I'll add that as a new JIRA ticket.

Rachel-Reverie replied:
For now I've added the current version requirements to the prerequisites and the check installed Python version sections.

kenwade4 commented on docs/guides/setup/installation/components_local/_check_python_version.mdx:
```diff
@@ -0,0 +1,21 @@
+### 1. Check Python version
+-->
+
+First, check that you have Python 3 installed. You can check your version of Python by running:
```
Maybe change `Python 3` to `Python 3.7+`

kenwade4 commented on docs/guides/setup/installation/components_local/_create_a_virtual_environment.mdx:
```diff
@@ -0,0 +1,27 @@
+
+<details>
+<summary>Python Virtual Environments</summary>
+We have chosen to use venv for virtual environments in this guide, because it is included with Python 3. You are not limited to using venv, and can just as easily install Great Expectations into virtual environments with tools such as virtualenv, pyenv, etc.
```
Do you want to mention the `conda` way? Also with pyenv, it installs specific versions of Python to use and doesn't manage virtual environments

kenwade4 replied:
Nevermind, there is a whole conda doc... maybe link to it from here?

Rachel-Reverie replied:
Rather than link to the section with the conda instructions, I've split the conda instructions out and included them in tabbed boxes under the headings with corresponding pip instructions.

kenwade4 commented on docs/guides/setup/installation/components_local/_create_a_virtual_environment.mdx:
````diff
@@ -0,0 +1,27 @@
+python3 -m venv venv
+```
+
+This command will create a new directory called `venv` where your virtual environment is located. In order to activate the virtual environment run:
````
Since this seems aimed at super beginners, it may be more clear to use a venv name other than venv (so they know the difference between the 2 venv in the command)

Rachel-Reverie replied:
I've pushed the corresponding update.

Rachel-Reverie committed:
DOC-340
- Specify supported versions of Python 3 in the prerequisites and check python version sections.
- Change venv name to `my_venv`

Rachel-Reverie committed:
DOC-340
- Split the Anaconda instructions out and place them in tabs under the headings with corresponding pip instructions

Rachel-Reverie committed:
Merge branch 'develop' into DOCS/DOC-280/DOC-340/partition_local_installation_guide

Rachel-Reverie committed:
DOC-340
- revert athena.md to its original state (somehow an edited version for another PR was included in this one.)

Rachel-Reverie committed:
Merge remote-tracking branch 'origin/DOCS/DOC-280/DOC-340/partition_local_installation_guide' into DOCS/DOC-280/DOC-340/partition_local_installation_guide

Rachel-Reverie committed:
DOC-340
- correct broken link in athena.md

kenwade4 commented on docs/guides/setup/installation/components_local/_create_a_virtual_environment.mdx:
```diff
@@ -0,0 +1,25 @@
+import CreateVenvPip from "./_create_an_venv_with_pip.mdx"
+import CreateVenvConda from "./_create_an_venv_with_anaconda.mdx"
+
+Our recommended best practice is to use a virtual environment for your Great Expectations project.  Both pip and Anaconda support the creation of virtual environments.
```
Maybe instead of "pip and Anaconda" put "standard Python 3 and Anaconda" since pip doesn't create venvs.

And maybe add something like "Once the virtual environment is created, you can install project requirements with `pip` or `conda`, depending on how the virtual environment was created."

Rachel-Reverie replied:
Right, that makes sense -- originally all of the instructions were divided between a pip and a conda workflow, and I overlooked that you aren't actually using pip to create the venv in that path.

Rachel-Reverie replied:
Right, that makes sense -- originally all of the instructions were divided between a pip and a conda workflow, and I overlooked that you aren't actually using pip to create the venv in that path.  I've updated things accordingly.

kenwade4 commented on docs/guides/setup/installation/components_local/_ensure_latest_pip.mdx:
```diff
@@ -0,0 +1,20 @@
+<!--
```
When this chunk is rendered, does the _create_an_venv_with_pip.mdx part get loaded right before? If not, you may want to mention to do the `source <venv-name>/bin/activate` before those python/python3 commands so they don't mess with the system install of python

Rachel-Reverie replied:
I've reorganzied things a bit to merge the sections where the version of pip/conda are checked with the section where the virtual environment is created, so that the pip update takes place after the virtual environment is created while the conda check remains prior.

Rachel-Reverie committed:
DOC-340
- correct information regarding creation of virtual environments.

Rachel-Reverie committed:
DOC-340
- reorganized some of the sections so that upgrading pip is done after activating a virtual environment, if that workflow is taken.

Rachel-Reverie committed:
Merge branch 'develop' into DOCS/DOC-280/DOC-340/partition_local_installation_guide

Rachel-Reverie committed:
Merge branch 'develop' into DOCS/DOC-280/DOC-340/partition_local_installation_guide
